### PR TITLE
Fix local initialisation of `IPFSTool/IPFSDaemon`

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR updates the IPFSDaemon to perform ipfs installation check only when initialised locally.

## Fix

https://github.com/valory-xyz/open-aea/pull/212#discussion_r914481029

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: plugins/aea-cli-ipfs/aea_cli_ipfs/ipfs_utils.py [hunk 1]
```diff
@@ -193,28 +204,23 @@ def __init__(self, addr: Optional[str] = None):
...
-        if addr is not None:
-            _ = resolve_addr(addr)  # verify addr
-            self._addr = addr
+        if addr is None:
+            addr = os.environ.get("OPEN_AEA_IPFS_ADDR", DEFAULT_IPFS_URL)
+
+        _, host, *_ = resolve_addr(addr)  # verify addr
 
+        self._addr = addr
+        self.is_remote = is_remote_addr(host)
```
Comment (angrybayblade) [inline plugins/aea-cli-ipfs/aea_cli_ipfs/ipfs_utils.py:213]: We check if the host is a remote address or local address.

File: plugins/aea-cli-ipfs/aea_cli_ipfs/ipfs_utils.py [hunk 2]
```diff
@@ -108,7 +116,10 @@ def __init__(self, node_url: str = "http://127.0.0.1:5001"):
         self.node_url = node_url
         self.api_url = node_url + IPFS_NODE_CHECK_ENDPOINT
         self.process = None
-        self._check_ipfs()
+        self.is_remote = is_remote
+
+        if not is_remote:
+            self._check_ipfs()
```
Comment (angrybayblade) [inline plugins/aea-cli-ipfs/aea_cli_ipfs/ipfs_utils.py:122]: We only perform check when initialised for local use.

File: plugins/aea-cli-ipfs/aea_cli_ipfs/ipfs_utils.py [hunk 3]
```diff
@@ -88,6 +88,11 @@ def addr_to_url(addr: str) -> str:
     return f"{protocol}://{host}:{port}"
 
 
+def is_remote_addr(host: str) -> bool:
+    """Check is addr is remote or local."""
```
Comment (Adamantios) [inline plugins/aea-cli-ipfs/aea_cli_ipfs/ipfs_utils.py:92]:
```suggestion
    """Check if addr is remote or local."""
```

**Review by Adamantios — COMMENTED** [review]
Why not keep the logic in the daemon instead of separating it in both the tool and the daemon?

**Review by dvilelaf — COMMENTED** [review]
LGTM

**Comment by angrybayblade** [comment]
> Why not keep the logic in the daemon instead of splitting it in both the tool and the daemon?

Which logic exactly?

**Comment by Adamantios** [comment]
> > Why not keep the logic in the daemon instead of splitting it in both the tool and the daemon?
> 
> Which logic exactly?

The newly added one, to check if the host is remote or not. I mean why pass around a flag to the daemon and not directly check in the daemon?

**Comment by angrybayblade** [comment]
> The newly added one, to check if the host is remote or not. I mean why pass around a flag to the daemon and not directly check in the daemon?

I think daemon class does not concern with checking if the host is remote or local. IPFSTool seems the right place to put the logic and `is_remote` variable.

**Review by Adamantios — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**